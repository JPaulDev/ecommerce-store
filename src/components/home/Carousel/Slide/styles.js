import styled from 'styled-components';

export const Container = styled.div`
  background-color: black;
  margin-right: -100%;
  min-height: 450px;
  overflow: hidden;
  padding: 10px 20px 0;
  position: relative;
  width: 100%;

  @media screen and (min-width: 550px) {
    padding: 10px 60px 0;
  }

  @media screen and (min-width: 715px) {
    padding: 90px 60px 0;
    min-height: 385px;
  }

  @media screen and (min-width: 1000px) {
    display: flex;
    justify-content: space-between;
  }
`;

export const Background = styled.div`
  position: absolute;
  right: -383px;
  top: 0;

  @media screen and (max-width: 714px) {
    &::after {
      background-image: linear-gradient(
        to right,
        rgb(0 0 0 / 40%),
        rgb(0 0 0 / 15%)
      );
      content: '';
      height: 100%;
      left: 0;
      position: absolute;
      width: 100%;
    }
  }

  @media screen and (min-width: 1000px) {
    right: initial;
    left: 50%;
    transform: translateX(-50%);
  }
`;

export const LeftContainer = styled.div`
  color: white;
  position: relative;
  text-align: center;

  @media screen and (min-width: 715px) {
    width: 40%;
    text-align: left;
  }

  @media screen and (min-width: 1000px) {
    width: 30%;
  }
`;

export const LeftLogoWrapper = styled.div`
  display: inline-block;
  margin-top: 10px;

  img {
    display: block;
    height: auto;
    max-width: 100%;
  }

  @media screen and (max-width: 714px) {
    width: 125px;
  }
`;

export const LeftHeading = styled.h4`
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.93rem;
  margin-bottom: 10px;
  margin-top: 5px;

  @media screen and (min-width: 715px) {
    margin-top: 15px;
  }
`;

export const Text = styled.p`
  font-size: 0.88rem;
  line-height: 1.6;
`;

export const CenterImageWrapper = styled.div`
  left: 50%;
  position: absolute;
  top: 180px;
  transform: translateX(-50%);
  width: 325px;

  img {
    display: block;
    height: auto;
    max-width: 100%;
  }

  @media screen and (min-width: 715px) {
    transform: translateX(-10%);
    top: 100px;
  }

  @media screen and (min-width: 850px) {
    width: 350px;
    top: 90px;
    right: 110px;
    left: initial;
    transform: initial;
  }

  @media screen and (min-width: 1000px) {
    width: 325px;
    top: 100px;
    left: 50%;
    transform: translateX(-50%);
  }

  @media screen and (min-width: 1100px) {
    width: 375px;
    top: 80px;
  }
`;

export const RightContainer = styled.div`
  color: ${({ textColor }) => textColor};
  position: relative;
  width: 30%;

  @media screen and (max-width: 999px) {
    display: none;
  }
`;

export const RightHeading = styled.h4`
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.3rem;
  font-weight: 600;
  margin-left: 10px;
  text-indent: 10px;

  img {
    margin-left: 15px;
  }
`;

export const List = styled.ul`
  font-size: 0.9rem;
  padding: 10px 0 0 18px;

  li {
    padding: 5px 0;
  }
`;
