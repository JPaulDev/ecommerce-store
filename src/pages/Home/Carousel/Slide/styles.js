import styled from 'styled-components';

export const Container = styled.div`
  background-image: linear-gradient(to right, #00000080, #00000030),
    url(${({ background }) => background});
  background-position: center, calc(100% + 383px) top;
  background-repeat: no-repeat;
  margin-right: -100%;
  min-height: 450px;
  padding: 10px 20px 0;
  position: relative;
  width: 100%;

  @media screen and (min-width: 550px) {
    padding: 10px 60px 0;
  }

  @media screen and (min-width: 715px) {
    padding: 90px 60px 0;
    min-height: 385px;
    background-image: url(${({ background }) => background});
    background-position: calc(100% + 383px) calc(50% + 30px);
  }

  @media screen and (min-width: 1000px) {
    display: flex;
    background-position: center calc(50% + 20px);
    justify-content: space-between;
  }
`;

export const LeftContainer = styled.div`
  color: white;
  text-align: center;

  @media screen and (min-width: 715px) {
    width: 40%;
    text-align: left;
  }

  @media screen and (min-width: 1000px) {
    width: 30%;
  }
`;

export const LeftLogo = styled.img`
  height: 15px;
  margin-top: 10px;
  width: 125px;

  @media screen and (min-width: 715px) {
    width: 158px;
    height: 19px;
  }
`;

export const LeftHeading = styled.h4`
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.93rem;
  margin-bottom: 10px;
  margin-top: 5px;

  @media screen and (min-width: 715px) {
    margin-top: 15px;
  }
`;

export const Text = styled.p`
  font-size: 0.88rem;
  line-height: 1.6;
`;

export const CenterImage = styled.img`
  height: 275px;
  left: 50%;
  position: absolute;
  top: 180px;
  transform: translateX(-50%);
  width: 325px;

  @media screen and (min-width: 715px) {
    transform: translateX(-10%);
    top: 100px;
  }

  @media screen and (min-width: 850px) {
    width: 350px;
    height: 296px;
    top: 90px;
    right: 110px;
    left: initial;
    transform: initial;
  }

  @media screen and (min-width: 1000px) {
    width: 325px;
    height: 275px;
    top: 100px;
    left: 50%;
    transform: translateX(-50%);
  }

  @media screen and (min-width: 1100px) {
    width: 375px;
    height: 317px;
    top: 80px;
  }
`;

export const RightContainer = styled.div`
  color: ${({ textColor }) => textColor || 'black'};
  width: 30%;
`;

export const RightHeading = styled.h4`
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.3rem;
  font-weight: 600;
  margin-left: 10px;
`;

export const RightLogo = styled.img`
  margin-left: 15px;
`;

export const List = styled.ul`
  font-size: 0.88rem;
  padding: 10px 0 0 18px;
`;

export const ListItem = styled.li`
  padding: 5px 0;
`;
