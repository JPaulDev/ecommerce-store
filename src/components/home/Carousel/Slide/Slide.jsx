import { forwardRef } from 'react';
import Image from 'next/future/image';
import * as Styled from './styles';

function Slide({ slideIndex, slide = {} }, ref) {
  return (
    <Styled.Container ref={ref} data-testid={`slide-${slideIndex + 1}`}>
      <Styled.Background>
        <Image src={slide.background} alt="" priority="true" quality={90} />
      </Styled.Background>
      <Styled.LeftContainer>
        <Styled.LeftLogoWrapper>
          <Image src={slide.leftLogo} alt={slide.leftLogoAlt} />
        </Styled.LeftLogoWrapper>
        <Styled.LeftHeading>{slide.leftHeading}</Styled.LeftHeading>
        <Styled.Text>{slide.leftText}</Styled.Text>
      </Styled.LeftContainer>
      <Styled.CenterImageWrapper>
        <Image
          src={slide.centerImage}
          alt={slide.centerImageAlt}
          quality={90}
        />
      </Styled.CenterImageWrapper>
      <Styled.RightContainer textColor={slide.textColor}>
        <Styled.RightHeading>
          {slide.rightHeading}
          <Image src={slide.rightLogo} alt={slide.rightLogoAlt} />
        </Styled.RightHeading>
        <Styled.List>
          {slide.rightText?.map((text) => (
            <li key={text}>{text}</li>
          ))}
        </Styled.List>
      </Styled.RightContainer>
    </Styled.Container>
  );
}

export default forwardRef(Slide);
