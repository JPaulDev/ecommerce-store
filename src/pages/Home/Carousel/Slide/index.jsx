import { forwardRef } from 'react';
import useMediaQuery from '../../../../hooks/useMediaQuery';
import * as Styled from './styles';

const Slide = forwardRef(({ slideIndex, slide }, ref) => {
  const isMatch = useMediaQuery('(min-width: 1000px)');

  return (
    <Styled.Container
      ref={ref}
      background={slide.background}
      data-testid={`slide-${slideIndex + 1}`}
    >
      <Styled.LeftContainer>
        <Styled.LeftLogo src={slide.leftLogo} alt={slide.leftLogoAlt} />
        <Styled.LeftHeading>{slide.leftHeading}</Styled.LeftHeading>
        <Styled.Text>{slide.leftText}</Styled.Text>
      </Styled.LeftContainer>
      <Styled.CenterImage src={slide.centerImage} alt={slide.centerImageAlt} />
      {isMatch && (
        <Styled.RightContainer textColor={slide.textColor}>
          <Styled.RightHeading>
            {slide.rightHeading}
            <Styled.RightLogo src={slide.rightLogo} alt={slide.rightLogoAlt} />
          </Styled.RightHeading>
          <Styled.List>
            {slide.rightText.map((text) => (
              <Styled.ListItem key={text}>{text}</Styled.ListItem>
            ))}
          </Styled.List>
        </Styled.RightContainer>
      )}
    </Styled.Container>
  );
});

export default Slide;
