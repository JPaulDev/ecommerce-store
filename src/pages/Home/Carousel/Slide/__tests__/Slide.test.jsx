import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Slide from '../index';

const slide = {
  topNavText: '30 SERIES 3XS PCs',
  background: '',
  centerImage: '',
  centerImageAlt: 'A PC with rainbow coloured LED case fans on the front',
  leftLogo: '',
  leftLogoAlt: '3XS Systems',
  leftHeading: 'RTXâ„¢ 30 Series PCs',
  leftText: `Custom built PCs built with NVIDIA's 2nd gen Ampere architecture,
      designed to deliver ultimate performance for gamers and creators.`,
  rightHeading: 'Powered by',
  rightLogo: '',
  rightLogoAlt: 'Nvidia GeForce RTX',
  rightText: [
    'Premium built to order systems',
    'Extensively tested',
    '7 day evening and weekend support',
    'Premium 3 year warranty',
    'Best in-class components',
    'Award-winning systems',
  ],
};

describe('Slide component', () => {
  it('renders a slide with all content present', () => {
    const tree = renderer
      .create(<Slide slide={slide} slideIndex={0} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Slide component at a screen width of 1000px and above', () => {
  it('renders the right text list', () => {
    window.resizeTo(1000, 768);
    render(<Slide slide={slide} slideIndex={0} />);
    const list = screen.getByRole('list');

    expect(list).toBeInTheDocument();
  });
});

describe('Slide component at a screen width of 999px and below', () => {
  it('hides the right text list', () => {
    window.resizeTo(999, 768);
    render(<Slide slide={slide} slideIndex={0} />);
    const list = screen.queryByRole('list');

    expect(list).toBeNull();
  });
});
