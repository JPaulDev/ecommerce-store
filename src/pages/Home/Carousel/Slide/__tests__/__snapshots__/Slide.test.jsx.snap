// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide component renders a slide with all content present 1`] = `
<div
  className="sc-bczRLJ VHwUl"
  data-testid="slide-1"
>
  <div
    className="sc-gsnTZi fEEGJC"
  >
    <img
      alt="3XS Systems"
      className="sc-dkzDqf jPkBsn"
      src=""
    />
    <h4
      className="sc-hKMtZM hXCOMb"
    >
      RTXâ„¢ 30 Series PCs
    </h4>
    <p
      className="sc-eCYdqJ kUimMJ"
    >
      Custom built PCs built with NVIDIA's 2nd gen Ampere architecture,
      designed to deliver ultimate performance for gamers and creators.
    </p>
  </div>
  <img
    alt="A PC with rainbow coloured LED case fans on the front"
    className="sc-jSMfEi hXysZZ"
    src=""
  />
  <div
    className="sc-gKXOVf fntncR"
  >
    <h4
      className="sc-iBkjds gOyMnu"
    >
      Powered by
      <img
        alt="Nvidia GeForce RTX"
        className="sc-ftvSup lnpREK"
        src=""
      />
    </h4>
    <ul
      className="sc-papXJ veWDa"
    >
      <li
        className="sc-jqUVSM lppNFK"
      >
        Premium built to order systems
      </li>
      <li
        className="sc-jqUVSM lppNFK"
      >
        Extensively tested
      </li>
      <li
        className="sc-jqUVSM lppNFK"
      >
        7 day evening and weekend support
      </li>
      <li
        className="sc-jqUVSM lppNFK"
      >
        Premium 3 year warranty
      </li>
      <li
        className="sc-jqUVSM lppNFK"
      >
        Best in-class components
      </li>
      <li
        className="sc-jqUVSM lppNFK"
      >
        Award-winning systems
      </li>
    </ul>
  </div>
</div>
`;
